packages/sanity/src/core/studio/colorScheme.tsx

```diff
@@ -20,25 +46,71 @@ export function ColorSchemeProvider({
   onSchemeChange,
   scheme: schemeProp,
 }: ColorSchemeProviderProps) {
+  const {getItem, setItem, removeItem} = useMemo(() => localStorageManager(), [])
+
+  // The system color scheme
   const prefersDark = usePrefersDark()
-  const [scheme, setScheme] = useState<ThemeColorSchemeKey>(schemeProp || 'light')
+  const systemScheme = prefersDark ? 'dark' : 'light'
 
-  // if the preferred color scheme changes, then react to this change
+  // The color scheme stored in local storage
+  const storedScheme = getItem?.(LOCAL_STORAGE_KEY) as ThemeColorSchemeKey | undefined
+
+  // A state indicating whether the user is using the system color scheme (mainly used to reflect what the user has selected in the UI)
+  const [usingSystemScheme, setUsingSystemScheme] = useState<boolean>(true)
+
+  // The initial scheme to use
+  const initialScheme = (schemeProp || storedScheme || systemScheme) as ThemeColorSchemeKey
```

Comment:
Wondering if `storedScheme` should be preferred over `schemeProp` here 🤔

Reply:
Hm, not sure. 🤔  The `StudioProvider` could be used isolation, and someone might have their own color scheme logic, and then I think you should be able to override "our logic" with the `schemeProp`